File: map/lab2/src/repository/ProjectValidator.java
package repository;

import domain.Project;

public class ProjectValidator implements Validator<Project> {
    @Override
    public void validate(Project pr) throws ValidationException {
        if (pr.getWeek() > 14 || pr.getWeek() < 1)
            throw new ValidationException("Week should be between 1 and 14 (inclusive).");
        if (pr.getId() < 0)
            throw new ValidationException("The id is invalid.");
    }
}

File: map/lab7/lab7/Domain/ProjectValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace lab7.Domain
{
    public class ProjectValidator : IValidator<Project>
    {
        public void Validate(Project pr)
        {
            var errorMsg = new List<string>();

            if (pr.Id < 0)
                errorMsg.Add("The id is invalid.");

            if (pr.Desc.Length < 2)
                errorMsg.Add("Description should contain at least 2 characters.");

            if (pr. Week > 14 || pr.Week < 1)
                errorMsg.Add("Week should be between 1 and 14 (inclusive).");

            if (errorMsg.Any())
                throw new ValidationException(String.Join("\n", errorMsg));
        }
    }
}